### PR TITLE
Open Source Record Store

### PR DESCRIPTION
**Pull request by es**
Final PR before open sourcing Record Store 🎉 
- Rename `recordstore` to `record_store`
- Update Gem description to reference latest Record Store
- Bump version to 1.0

cc: @Shopify/dns

### CODE REVIEW

**Comment by wvanbergen** [comment]
:+1::tada::heart:

**Comment by dwradcliffe** [comment]
Fantastic!

File: README.md [hunk 1]
```diff
@@ -83,15 +83,15 @@ To get started developing on Record Store, run `bin/setup`. This will create a d
 
 ### Adding new Providers
 
-To add a new Provider, create a class inherriting `Provider` in [`lib/recordstore/provider/`](lib/recordstore/provider/). The [DynECT provider](lib/recordstore/provider/dnsimple.rb) is good to use as a reference implementation.
+To add a new Provider, create a class inherriting `Provider` in [`lib/record_store/provider/`](lib/record_store/provider/). The [DynECT provider](lib/record_store/provider/dnsimple.rb) is good to use as a reference implementation.
```
Comment (aeden) [inline README.md]: Should that be "DNSimple provider" or should the file path be changed?

**Comment by n1koo** [comment]
Awesome @es !

File: README.md (hunk 1 above)
Comment (es) [inline README.md]: Good catch, it should be a different file path. DynECT was the original implementation and doesn't have as much data shaping.

**Pull request merged**